packages/components/src/device-switcher/fixed-viewport.tsx

```diff
@@ -0,0 +1,61 @@
+import { CSSProperties } from 'react';
+
+// Device viewport width in pixels
+const DEVICE_COMPUTER_WIDTH = 1080;
+const DEVICE_TABLET_WIDTH = 782;
+const DEVICE_PHONE_WIDTH = 480;
+
+const deviceWidthById = {
+	computer: DEVICE_COMPUTER_WIDTH,
+	tablet: DEVICE_TABLET_WIDTH,
+	phone: DEVICE_PHONE_WIDTH,
+} as Record< string, number >;
+
+export const useViewportScale = ( device: string, viewportWidth: number ) => {
+	const deviceWidth: number = deviceWidthById[ device ];
+	let width = viewportWidth;
+
+	if ( ! viewportWidth ) {
+		// Scale is 1 when the feature is disabled
+		return 1;
+	}
+
+	if ( 'computer' !== device && viewportWidth > deviceWidth ) {
+		// Use device width as max width for tablet and phone
+		width = deviceWidth;
+	}
+
+	return width / deviceWidth;
+};
+
+interface Props {
+	children: React.ReactNode;
+	frameRef?: React.MutableRefObject< HTMLDivElement | null >;
+	device: string;
+	enabled?: boolean;
+}
+
+const FixedViewport = ( { children, frameRef, device, enabled }: Props ) => {
+	const viewportWidth = frameRef?.current?.parentElement?.clientWidth as number;
+	const viewportScale = useViewportScale( device, viewportWidth );
+
+	if ( ! enabled ) {
+		return <>{ children }</>;
```

Comment:
The fragment is required to avoid this error, any other ideas?

<img width="713" alt="Screenshot 2566-06-12 at 15 36 10" src="https://github.com/Automattic/wp-calypso/assets/1881481/c45c019d-3fc5-4532-b6ce-3296b5751130">

Reply:
I did a change so we don't need it. WDYT?